packages/svelte/src/internal/client/dom/template.js

```diff
@@ -33,6 +34,30 @@ export function push_template_node(
 	return dom;
 }
 
+/**
+ * @param {string} filename
+ * @param {string} filename_hash
+ */
+function attach_inspector_metadata(filename, filename_hash) {
+	const nodes = document.querySelectorAll(`*[${filename_hash}]`);
```

Comment:
This is going to get called for every block in every instance of every component — even though we're removing the attribute the first time around, that's a _lot_ of `document.querySelectorAll`, especially during hydration when it all happens at once. Dev-only or not, are we concerned about the overhead of this?

Reply:
In benchmarked it, on a document with 100k elements, this takes just over a ms. The more you run it, the faster it gets too. Running it a second time, it goes to 0.56ms for me.